mathics/eval/rules.py

```diff
@@ -0,0 +1,118 @@
+from typing import List, Optional as OptionalType, Tuple, Union
+
+from mathics.core.atoms import String
+from mathics.core.element import BaseElement
+from mathics.core.evaluation import Evaluation
+from mathics.core.expression import Expression
+from mathics.core.list import ListExpression
+from mathics.core.rules import Rule
+from mathics.core.symbols import Atom, Symbol, SymbolList
+from mathics.core.systemsymbols import SymbolDispatch, SymbolRule, SymbolRuleDelayed
+
+
+# TODO: disentangle me
+def create_rules(
```

Comment:
Please name `eval_Replace`. This is used only in Replace* built-in functions. In other places, an xxxList  builtin is assumed to call the xxx eval underneath (over a list)

Reply:
About this, I am not totally sure: this function is supposed to convert `Expression` objects into (Python's list of) `Rule`s objects. The call to the `apply` method is done elsewhere.